packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1391,6 +1412,28 @@ abstract class ResidentRunner {
 
   // Clears the screen.
   void clearScreen() => globals.logger.clear();
+
+  Future<vm_service.Isolate> waitForExtension(vm_service.VmService vmService, String extension) async {
+    final Completer<void> completer = Completer<void>();
+    try {
+      await vmService.streamListen(vm_service.EventStreams.kExtension);
+    } on Exception {
+      // do nothing
+    }
+    vmService.onExtensionEvent.listen((vm_service.Event event) {
+      if (event.json['extensionKind'] == 'Flutter.FrameworkInitialization') {
+        completer.complete();
```

Comment:
It looks like there can be more than one of these.

Reply:
See - https://github.com/flutter/flutter/issues/74407. Working on a fix now. Have the fix - just working on the test.